src/crypto/Signature.ts

```diff
@@ -248,14 +302,107 @@ export interface AccountLinkSigner {
   ): Promise<AccountLinkedPublicKey>
 }
 
+export class SIWEWalletAccountLinkSigner implements AccountLinkSigner {
+  wallet: Signer
+
+  constructor(wallet: Signer) {
+    this.wallet = wallet
+  }
+
+  public static accountLinkedSIWERoleRequestText(
+    role: AccountLinkedRole
+  ): string {
+    switch (role) {
+      case AccountLinkedRole.INBOX_KEY:
+        return 'AllowAllRead'
+      case AccountLinkedRole.SEND_KEY:
+        return 'GrantSendPermissions'
+    }
+  }
+
+  public static accountLinkedSIWEResourceRoleText(
+    keyBytes: Uint8Array,
+    role: AccountLinkedRole
+  ): string {
+    return `https://xmtp.org/siwe/${SIWEWalletAccountLinkSigner.accountLinkedSIWERoleRequestText(
+      role
+    )}/secp256k1/${bytesToHex(keyBytes)}`
+  }
+
+  // Default SIWE text to be signed, most apps will NOT want to use this.
+  // Most apps will want to include the resource string in their own SIWE to avoid
+  // asking for multiple SIWE signatures.
+  public static defaultAccountLinkSIWERequestText(
+    key: UnsignedPublicKey,
+    role: AccountLinkedRole,
+    walletAddress: string
+  ): string {
+    // Create a SIWE message
+    // - statement can be anything
+    // - get the address from the signer
+    // - add the resource string with keybytes
+    const keyBytes = key.toBytes()
+    const resource =
+      SIWEWalletAccountLinkSigner.accountLinkedSIWEResourceRoleText(
+        keyBytes,
+        role
+      )
+    const siwe = new SiweMessage({
+      statement: 'XMTP Account Link with Role: ' + role,
+      address: walletAddress,
+      domain: 'xmtp.org',
```

Comment:
What would it mean for a user to sign a "GrantSendPermissions" from xmtp.org?

Reply:
Not much at the moment, but we need a domain placeholder to satisfy the SIWE format. I think I mention in the comment but this illustrates the flow or provides a default SIWE message. Main goal of this PR is to add the code that can validate a SIWE signature